crypto/s2n_tls13_keys.c

```diff
@@ -186,7 +186,7 @@ int s2n_tls13_derive_early_secrets(struct s2n_tls13_keys *keys, struct s2n_psk *
     } else {
         /* Sanity check that an early secret exists */
         ne_check(psk->early_secret.size, 0);
-        keys->extract_secret = psk->early_secret;
+        memcpy_check(keys->extract_secret.data, psk->early_secret.data, psk->early_secret.size);
```

Comment:
This is only correct if psk->early_secret.size == keys->extract_secret.size. We should probably assert that instead of that the early secret isn't empty.

Reply:
This assert should be `lte_check(psk->early_secret.size, keys->extract_secret.size);`